packages/contracts/src/dollar/libraries/LibBondingCurve.sol

```diff
@@ -0,0 +1,235 @@
+    }
+
+    function bondingCurveStorage() internal pure returns (BondingCurveData storage l) {
+        bytes32 slot = BONDING_CONTROL_STORAGE_SLOT;
+        assembly {
+            l.slot := slot
+        }
+    }
+
+    function setParams(
+        uint32 _connectorWeight, 
+        uint256 _baseY
+    ) internal {
+        require(_connectorWeight > 0 && _connectorWeight <= 1000000, "invalid values"); 
+        require(_baseY > 0, "must valid baseY");
+
+        bondingCurveStorage().connectorWeight = _connectorWeight;
+        bondingCurveStorage().baseY = _baseY;
+        emit ParamsSet(_connectorWeight, _baseY);
+    }
+
+    function connectorWeight() internal returns (uint32) {
+        return bondingCurveStorage().connectorWeight;
+    }
+
+    function baseY() internal returns (uint256) {
+        return bondingCurveStorage().baseY;
+    }
+    
+    function poolBalance() internal returns (uint256) {
+        return bondingCurveStorage().poolBalance;
+    }
+
+    function deposit(uint256 _collateralDeposited, address _recipient)
+        internal
+    {
+        BondingCurveData storage ss = bondingCurveStorage();
+        require(
+            ss.connectorWeight != 0 && ss.baseY != 0, "not set"
+        );
+
+        uint256 tokensReturned;
+
+        if (ss.tokenIds > 0) {
+            tokensReturned = purchaseTargetAmount(
+                _collateralDeposited,
+                ss.connectorWeight,
+                ss.tokenIds,
+                ss.poolBalance
+            );
+        } else {
+            tokensReturned = purchaseTargetAmountFromZero(
+                _collateralDeposited,
+                ss.connectorWeight,
+                ACCURACY,
+                ss.baseY
+            );
+        }
+
+        IERC20 dollar = IERC20(
+            LibAppStorage.appStorage().dollarTokenAddress
+        );
+        dollar.transferFrom(
+            _recipient,
+            address(this),
+            _collateralDeposited
+        );
+
+        ss.poolBalance = ss.poolBalance + _collateralDeposited;
+        bytes memory tokReturned = toBytes(tokensReturned);
+        ss.share[_recipient] += tokensReturned;
+        ss.tokenIds += 1;
+
+        IERC1155Ubiquity bNFT = IERC1155Ubiquity(
+            LibAppStorage.appStorage().ubiquiStickAddress
+        );
+        bNFT.mint(
+            _recipient, 
+            ss.tokenIds, 
+            tokensReturned, 
+            tokReturned 
+        );
+
+        emit Deposit(_recipient, _collateralDeposited);
+    }
+
+    function getShare(address _recipient) internal returns (uint256) {
+        BondingCurveData storage ss = bondingCurveStorage();
+        return ss.share[_recipient];
+    }
+
+    function toBytes(uint256 x) internal pure returns (bytes memory b) {
+        b = new bytes(32);
+        assembly { mstore(add(b, 32), x) }
+    }
+
+    function withdraw(uint256 _amount) internal {
+        BondingCurveData storage ss = bondingCurveStorage();
```

Comment:
Why is this variable named `ss` is this forked from our old staking shares code?

Reply:
No it's not forked. Just following the diamond facet schema.

Reply:
What does `ss` stand for?

Reply:
I just randomly picked it following the patter of other facets. I could have equally called it "bs", "bonding storage"